app/client/components/LinkingState.ts

```diff
@@ -54,105 +91,150 @@ function isSummaryOf(summary: TableRec, detail: TableRec): boolean {
-      // inverse of summary-table's 'group' column, we could implement both, and more efficiently.
-      const isDirectSummary = srcSection.table().summarySourceTable() === tgtSection.table().getRowId();
-      const _filterColValues = ko.observable<FilterColValues>();
-      this.filterColValues = this.autoDispose(ko.computed(() => _filterColValues()));
+      // !srcCol && !tgtCol && summary => typical summary filter-linking
+      // We do summary filtering if no cols specified and summary section is linked to a more detailed summary
+      // (or to the summarySource table)
 
       // source data table could still be loading (this could happen after changing the group by
       // columns of a linked summary table for instance), hence the below listener.
+      const _filterState = ko.observable<FilterState>();
+      this.filterState = this.autoDispose(ko.computed(() => _filterState()));
       this.autoDispose(srcTableData.dataLoadedEmitter.addListener(_update));
 
+      const self = this;
       _update();
       function _update() {
-        const result: FilterColValues = {filters: {}, operations: {}};
-        if (srcSection.isDisposed()) {
-          return result;
+        if (srcSection.isDisposed() || srcSection.table().groupByColumns().length == 0) {
+          //srcSection disposed can happen transiently while loading? (unsure)
+          //groupByColumns == [] can happen if we make a summary tab [group by nothing]. (in which case: don't filter)
+          _filterState(EmptyFilterState);
+          return;
         }
-        const srcRowId = srcSection.activeRowId();
-        for (const c of srcSection.table().groupByColumns()) {
-          const colId = c.colId();
-          const srcValue = srcTableData.getValue(srcRowId as number, colId);
-          result.filters[colId] = [srcValue];
-          result.operations[colId] = 'in';
-          if (isDirectSummary && isListType(c.summarySource().type())) {
-            // If the source groupby column is a ChoiceList or RefList, then null or '' in the summary table
-            // should match against an empty list in the source table.
-            result.operations[colId] = srcValue ? 'intersects' : 'empty';
-          }
+
+        //Make one filter for each groupBycolumn of srcSection
+        const resultFilters: (ko.Computed<FilterState>|undefined)[] = srcSection.table().groupByColumns().map(srcGCol =>
+          self._makeFilterObs(srcGCol, summaryGetCorrespondingCol(srcGCol, tgtSection.table()))
+        );
+
+        //If any are null, error out
+        if(resultFilters.filter((f) => f == undefined).length > 0) {
+          console.warn("LINKINGSTATE: some of filters are null", resultFilters);
+          _filterState(EmptyFilterState);
+          return;
         }
-        _filterColValues(result);
-      }
+
+        //Merge them together in a computed
+        const resultComputed = self.autoDispose(ko.computed(() => {
```

Comment:
The computed is necessary I think. the FilterState for a summary table should look like `{filters: {colA: [val], colB: [val], colC:[val]} ...}`
each makeFilterObs returns an `Observable<FilterState>`  for a single column `filters: {colA:[val]}` at a time, so we need the computed to merge them all together into a single FilterState for all the columns simulatenously.

You're right that the resultComputed won't be disposed, I need to fix that. There's also the issue that the makeFilterObs won't be disposed either, and I'm not sure how to handle that with holders since there's going to be many of them.

Reply:
But, won't it work the same if you just merge them outside the computed?

Reply:
There's no way to access them outside the computed. They're only created in _update. The only way info gets out of _update is by being pushed into the _filterstate observable, for which we need to merge them first

I guess I could push that outside, but it would require an extra observable array outside the _update function to push them into